### DISCUSSION TITLE
download less

### DISCUSSION THREAD

**Discussion by philipheinser** [coollabsio/coolify#4611] [Improvement Requests]
would it make sense to add `--depth 1` to download only last commit?

https://github.com/coollabsio/coolify/blob/aae54e8847efdd6ccab553d24717c863030c9eac/app/Models/Application.php#L1072

**Comment by wizzyto12** [comment]
I agree. We have a very big monorepo, which takes more than 2 minutes to clone. This would be a really nice build time-saver!

**Comment by sina-ss** [comment]
is there any update or someone doing that?

**Comment by sina-ss** [comment]
I'm still waiting. When ever our team deploy new version, server downloads about 1GB!

**Comment by margot-vogo** [comment]
hi folks,
when using --depth 1 after ssh address git repository git copy downloading only last commit

tadaaaa - is done :)

<img width="1134" height="294" alt="image" src="https://github.com/user-attachments/assets/5767574a-d169-4769-bb01-9bd4d96ae00b" />

**Reply by checkerschaf** [reply]
Hey, that doesn't work for me on latest `v4.0.0-beta.420.6`. I'm using a connected Git source, as the repo is private. 

```
Oops something is not okay, are you okay? 😢
GitHub API call failed:<br>Error: Not Found<br>Rate Limit Status:<br>- Remaining Calls: 4999<br>- Reset Time: 2025-08-12 10:25:20 UTC
Deployment failed. Removing the new version of your application.
```

I think it it's replaced like this for private repos: `https://github.com/git@github.com:username/repo.git%20--depth%201/commits/main`
